### ISSUE TITLE
[Bug]: Skyrocketing memory usage, seemingly unprompted, with disk utilization to match upon closing

### ISSUE THREAD

**Issue by daria-mianne** [MelvorIdle/melvoridle.github.io#1723]
### Describe the bug

While watching a youtube video with Melvor not the active window (but also not minimized), I noticed that my computer suddenly started stuttering badly. I opened the task manager, which took multiple attempts and almost a minute of waiting, and found that, while the system's overall RAM utilization was listed as being at 99%, no applications were reported as being particularly large contributors to this. At first, I thought it might be Firefox's issue, as I had multiple windows open and two of them had pretty high tab counts, but closing one of these windows actually increased the reported memory usage of Firefox from 800MB to 1.8GB. At the same time, Melvor's reported memory usage began quickly and steadily climbing, gaining roughly 2GB/minute.

Once the game's reported memory usage reached 8GB, I attempted to close it. However, instead of fixing the issue, the reported memory's increases continued unabated, and disk utilization shot up to 100%. At this time, the game was reported as using approximately 40MB/s of disk. After a couple more minutes of amazement and wondering whether it would stop on its own, the reported memory usage climbed to 20GB and the disk usage spiked even higher to 188.8MB/s.

At this time, I decided to kill the program via the task manager. The computer hanged for a very brief moment, and then the mouse became a spinner. After several seconds of this, the program closed and the computer's memory and disk usage fell to about 20% and 0%, respectively. By the time it closed, Melvor was using about 22GB of RAM.

I have since re-opened the game, and system memory usage has only gone back up to 23%.

I have two mods installed - Melvor Auto Farming and Melvor Action Queue. The former is being used to automatically farm strawberries, barrentoes, and maples, while the latter is only being used for its automatic mastery pool management with no actual usage of its action queueing.

### Reproduction Steps

Unknown. This occurred without user input while the Melvor window was open in the background, performing Astrology research on the first constellation, Deedree. My mastery pool for the skill was roughly 49%, my mastery for Deedree was level 96, and I had not begun research into any other constellations.

### Expected behaviour

Memory and disk utilization should not even be close to maxing out a system with 32GB of RAM and an nVME SSD.

### Screenshots

Memory column from task manager before closing anything: 
![image](https://user-images.githubusercontent.com/29166401/142291407-5e17794a-f4f5-4c66-ab66-5802920104ce.png)

Full line from the task manager not long after attempting to close the game: ![image](https://user-images.githubusercontent.com/29166401/142291100-bcb14046-4ce7-415a-b50f-d6c87cdfa053.png)

Disk usage spike later in the close process: ![image](https://user-images.githubusercontent.com/29166401/142291112-fee10654-8ca7-452b-8c9b-587cad6ed882.png)

### Which platforms are you experiencing this bug on?

- [ ] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Safari
- [ ] Mobile App (iOS)
- [ ] Mobile App (Android)
- [X] Steam
- [ ] Other (Please Specify)

### Which version of the game are you experiencing this bug on?

Beta v1.0

### Game Subversion

?1329

### Are you using any scripts or extensions?

Yes

**Comment by daria-mianne** [comment]
A clarification I included in discord but neglected to mention here:

I don't think it's actually using more memory now than it was before I noticed. I think that windows is gradually reporting more and more accurately on the amount of memory it's using, as the percentage of total memory in use had actually gone down in that time, from 99% to 93%.

**Comment by MelvorIdle** [comment]
One thing that I'd need to be sure the game is causing this, is confirmation that this does not occur when those scripts are disabled.

Although it may very well be the game, its something that is quite important when debugging issues like this.

**Comment by daria-mianne** [comment]
I'd include that information if I had it, but for now, it's only happened the one time and I have no idea how to reproduce it. I'll disable them for a while and let you know if it happens again.

**Comment by daria-mianne** [comment]
I haven't been able to allow the game to be open at all times in the time since my last comment, but it has been open for more than 48 hours currently and the memory leak hasn't reoccurred yet. I've been watching Mevlor's RAM consumption in the task manager for the last hour, and the memory has risen and fallen between 1090MB and 1190MB in a saw pattern - it increase gradually and then drops 100MB instantly, with a period of somewhere between 5 and 10 minutes. This behavior makes sense for garbage collection in my opinion, so I doubt the game is going to run into the issue again as-is.

These observations were made in a slightly different setup than the original filing:
- The mods are disabled
- The total mastery pool is above 50% at any given time
- I've leveled Deedree and Iridan to 99 mastery, Ameria to 94, Terra to 74, and the rest to 30

I'll next reenable the mods and see if the memory grows over time, rather than cycling between two consistent values for min and max. Hopefully, it was just a 1-time fluke, but if not, I'll then move on to investigate which mod is causing the issue. Either way, I believe it's safe to close this issue at your discretion, as it seems unlikely that any of your code is at fault.

**Comment by MelvorIdle** [comment]
Thanks for the detailed response and follow up!

I will leave this open just in case something else comes up.

**Comment by AlexDerr** [comment]
I have noticed this as well but mine is happening after the game closes and seems to be a memory leak.

Twice now several hours after closing the game I have looked in my task manager to see melvor idle taking ~26GB of ram.

I am using the steam client, no mods. V1.0 ?1352

All that I am doing in the game right now is logging on, allocating astrology extra mastery xp, doing farming and logging out. I am currently training Syllia and putting xp into all others below it. With that being all that I am doing in the game I don't have any extra info on what could be causing it or reproduction steps unfortunately
